.github/workflows/docs.yml

```diff
@@ -0,0 +1,48 @@
+name: Deploy Documentation
+
+on:
+  workflow_dispatch:
+
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        config-file: docs/mlc-config.json
+        folder-path: docs
+
+  build-and-deploy-docs:
+    needs: [markdown-link-check]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python Environment
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          pip install --upgrade pip
+
+      - name: Install Documentation Dependencies
+        run: |
+          source venv/bin/activate
+          pip install -r docs/requirements.txt
+
+      - name: Build documentation
+        run: |
+          source venv/bin/activate
+          mkdocs build
+
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          folder: site
+          branch: gh-pages
+          git-config-name: "Github Action"
+          commit-message: "doc update for tag `${{ github.event.client_payload.tag || github.ref_name }}`"
```

Comment:
NIT, please add an empty line at the end. Sometimes it messes things up or leads to additional changes if an Editorconfig gets introduced at some point.

Reply:
Fixed in [843dfb6](https://github.com/unitycatalog/unitycatalog/pull/677/commits/843dfb6e441d3e635954cdd9cbaa6427bf9e47c6)